### PR TITLE
Give receptor.conf write permissions for awx

### PR DESCRIPTION
**Pull request by fosterseth**
##### SUMMARY

related https://github.com/ansible/awx-operator/issues/1094

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

### CODE REVIEW

**Review by rooftopcellist — APPROVED** [review]

**Comment by TheRealHaoLiu** [comment]
i dont think we can run chmod because we are not running this as root (also should not run it as root)

alternative suggestion: 
we can do the change in the dockerfile for the awx-ee image and change `cp` to `cat >` to make sure we preserve permission

File: roles/installer/templates/deployments/deployment.yaml.j2 [hunk 1]
```diff
@@ -317,6 +317,8 @@ spec:
               if [ ! -f /etc/receptor/receptor.conf ]; then
                 cp /etc/receptor/receptor-default.conf /etc/receptor/receptor.conf
                 sed -i "s/HOSTNAME/$HOSTNAME/g" /etc/receptor/receptor.conf
+                chown awx /etc/receptor/receptor.conf
+                chmod 664 /etc/receptor/receptor.conf
```
Comment (rooftopcellist) [inline roles/installer/templates/deployments/deployment.yaml.j2]: We should actually do this in the Dockerfile instead, the cat the receptor.conf into the "stub" receptor.conf file, like we do for rsyslog.conf.

Comment (TheRealHaoLiu) [reply roles/installer/templates/deployments/deployment.yaml.j2]: at least we can't do chown since awx is not a user that exist

Comment (TheRealHaoLiu) [reply roles/installer/templates/deployments/deployment.yaml.j2]: i think just the chmod will work but i cant reproduce this issue on my openshift cluster

Comment (TheRealHaoLiu) [reply roles/installer/templates/deployments/deployment.yaml.j2]: /etc/receptor is a emptyDir mount so when task container and web container mount the emptyDir it should be own by them so ownership shouldnt be a issue

**Review by rooftopcellist — APPROVED** [review]

**Comment by shanemcd** [comment]
@fosterseth Have you reproduced the original error? I kinda doubt this will be sufficient.

**Comment by fosterseth** [comment]
@shanemcd it might be a false flag -- we introduced some incompatibility with the execution capacity expansion feature 

awx 21.6+ requires awx operator 0.29.0+

I noticed the user who opened the issue is on 0.28. I asked them to try on later operator versions

File: roles/installer/templates/deployments/deployment.yaml.j2 (hunk 1 above)
Comment (timansky) [reply roles/installer/templates/deployments/deployment.yaml.j2]: The Pod Security Context supports setting an fsGroup, which allows you to set the group ID that owns the volume, and thus who can write to it.

[More](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/)

**Comment by fosterseth** [comment]
I don't think this is needed anymore

**Pull request closed without merging**